document/CoreFormat.mdk

```diff
@@ -355,12 +359,52 @@ An `issue` object describes a single issue detected by a static analysis tool.
 ### `ruleId` property { #issue-ruleId }
 
 An `issue` object shall contain a property named `ruleId` whose value is a string containing
-a stable, opaque identifier for the rule that was violated.
+the stable, opaque identifier for the rule that was violated.
 
 ~ IsoExample
 `"CA2101"`
 ~
 
+### `kind` property { #issue-kind }
+
+An `issue` object may contain a property named `kind` whose value is one of a fixed set of strings
+that specify what kind of issue this is.
+
+If present, the `kind` property shall have one of the following values, with the specified meanings:
+
+* `warning`: The rule specified by the `ruleId` property (&sect;[#issue-ruleId]) was evaluated and a violation was found.
+
+* `error`: The rule specified by the `ruleId` property was evaluated and a serious violation was found.
+
+* `pending`: The rule specified by the `ruleId` property was evaluated, but whether the result of the evaluation
+constitutes a violation cannot be determined without further evaluation, either by a person or by an automated tool.
+
+~ IsoNote
+In some cases, team policy or corporate policy might determine if an analysis result constitutes a violation.
+~
+
+* `internalError`: The tool encountered an internal error that prevented it from performing a complete analysis.
+
+    If the error occurred in the course of evaluating a specific rule, the `ruleId` property should specify that rule.
+
+    If the error did not occur in the course of evaluating a specific rule, the `ruleId` property should be set
+    to a value distinct from all other rules.
+
+~ IsoNote
+Depending on the nature of the error, the tool might or might not be able to continue running.
+In all cases, the presence of an `issue` object whose `kind` property is set to `internalError` indicates
+that the run failed, because it could could not successfully run all specified rules on all specified
+analysis targets.
+
+If a tool encounters an error in the course of evaluating a rule, the tool might report the error, disable the rule,
+and continue to execute the remaining rules.
+
+If a tool encounters an error outside of the evaluation of a rule, the tool might report
+the error to the log and halt. (The tool might also just exit, either normally or abnormally.)
+~
+
+If this property is not present, it shall be considered equivalent to the value `warning`.
```

Comment:
Can we say explicitly 'if the 'kind' property is not present, it shall be considered'. i'd have to look at the doc with all formatting, but reading through the markdown, it seems like we've traveled a ways since the last reference to kind, so 'this property' is not as immediately obvious.